src/contrib/images.jl

```diff
@@ -0,0 +1,94 @@
+#=
+ImageTransformations
+=#
+# function warp(img::AstroImageMat, args...; kwargs...)
+#     out = warp(arraydatat(img), args...; kwargs...)
+#     return copyheaders(img, out)
+# end
+
+
+
+# Instead of using a datatype like N0f32 to interpret integers as fixed point values in [0,1],
+# we use a mappedarray to map the native data range (regardless of type) to [0,1]
+ImageCore.normedview(img::AstroImageMat{<:FixedPoint}) = img
+function ImageCore.normedview(img::AstroImageMat{T}) where T
+    imgmin, imgmax = extrema(skipmissingnan(img))
+    Δ = abs(imgmax - imgmin)
+    normeddata = mappedarray(
+        pix -> (pix - imgmin)/Δ,
+        pix_norm -> convert(T, pix_norm*Δ + imgmin),
+        img
+    )
+    return shareheader(img, normeddata)
+end
+
+"""
+    clampednormedview(arr, (min, max))
+
+Given an AbstractArray and limits `min,max` return a view of the array
+where data between [min, max] are scaled to [0, 1] and datat outside that
+range are clamped to [0, 1].
+
+See also: normedview
+"""
+function clampednormedview(img::AbstractArray{T}, lims) where T
+    imgmin, imgmax = lims
+    Δ = imgmax - imgmin
+    # Do not introduce NaNs if colorlimits are identical
+    if Δ == false
+        Δ = true
+    end
+    normeddata = mappedarray(
+        pix -> clamp((pix - imgmin)/Δ, false, true),
+        img
+    )
+    return maybe_shareheader(img, normeddata)
+end
+function clampednormedview(img::AbstractArray{T}, lims) where T <: Normed
+    # If the data is in a Normed type and the limits are [0,1] then
+    # it already lies in that range.
+    if lims[1] == 0 && lims[2] == 1
+        return img
+    end
+    imgmin, imgmax = lims
+    Δ = imgmax - imgmin
+    # Do not introduce NaNs if colorlimits are identical
+    if Δ == false
+        Δ = true
+    end
```

Comment:
I'm confused.  Do I understand correctly `Δ` is the difference between two numbers?  How can it be `== false`?  Also, side note, I always find it weird checks `if x == false` or `if x == true`, I usually do `if !x` or `if x`, which also makes sure `x` is a boolean.

Reply:
Thanks for spotting this. I sometimes use true and false as numerical values since they promote nicely. Agreed this should just be `== 0`